module-4/backend/week_three.md

```diff
@@ -1,19 +1,66 @@
 ## Week Three - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+1. At a high level, what is Node?
 
-Note: When you're done, submit a PR. 
+It's a server side runtime environment for javascript. 
 
-1. At a high level, what is Node?
 2. What is Express? What is Express similar to in the Ruby world?
+
+Express is a node web app framework; Express is to JS as Sinatra is to Ruby in terms of provided functionality. 
+
 3. How do we setup a route when creating an API with Node and Express? Please provide a code snippet.
+// HTTP verb + URL + controller + action
+```app.get('/api/v1/foods', FoodsController.indexFood)
+app.post('/api/v1/foods', FoodsController.createFood)
+app.put('/api/v1/foods/:id', FoodsController.editFood)
+app.delete('/api/v1/foods/:id', FoodsController.deleteFood)```
+
+*if a controller and model objects not created yet, the second argument would be a function(request, response){} to handle the request and response data
+
 4. What do we use Knex for?
+
+It is the ORM for node/js. It allows us to connect to and communicate with a databbase by making queries. 
+
 5. How could you organize your code to follow the MVC design pattern in your Quantified Self project?
+
+By abstracting the code for setting up an API endpoint into separate files with separate responsibilities: 
+
+A: model level functions for SQL queries; layer between JS server and datababse server
+
+B: controller layer: layer between model objects and router; responsible to perform different actions on POJOs by handling the 
+request and response data
+
+C: routes: specify HTTP verb and pass URL and controller object function as arguments
+
+Example: ```app.get('/api/v1/foods', FoodsController.indexFood)``` demosntrates Rails like routing pattern where a verb and URL are routed to a controller and a specific action within that controller
```

Comment:
i really like this example! just a small note though, since we're calling `indexFood` on `FoodsController`, we probably don't need to specify that it's an index for food - `FoodsController.index` is probably explicit enough and avoids redundancy 👍